heron/executor/src/python/heron_executor.py

```diff
@@ -125,27 +129,31 @@ class HeronExecutor(object):
   container. Based on the container id and the instance distribution, it determines if the container
   is a master node or a worker node and it starts processes accordingly."""
   def __init__(self, args, shell_env):
+    parsed_args = self.parse_args(args)
+
     self.shell_env = shell_env
     self.max_runs = 100
     self.interval_between_runs = 10
-    self.shard = int(args[1])
-    self.topology_name = args[2]
-    self.topology_id = args[3]
-    self.topology_defn_file = args[4]
-    self.zknode = args[5]
-    self.zkroot = args[6]
-    self.tmaster_binary = args[7]
-    self.stmgr_binary = args[8]
-    self.metricsmgr_classpath = args[9]
+    self.shard = int(parsed_args.shard)
+    self.topology_name = parsed_args.topology_name
+    self.topology_id = parsed_args.topology_id
+    self.topology_defn_file = parsed_args.topology_defn_file
+    self.zknode = parsed_args.zknode
+    self.zkroot = parsed_args.zkroot
+    self.tmaster_binary = parsed_args.tmaster_binary
+    self.stmgr_binary = parsed_args.stmgr_binary
+    self.metricsmgr_classpath = parsed_args.metricsmgr_classpath
     self.instance_jvm_opts =\
-        base64.b64decode(args[10].lstrip('"').rstrip('"').replace('&equals;', '='))
-    self.classpath = args[11]
-    self.master_port = args[12]
-    self.tmaster_controller_port = args[13]
-    self.tmaster_stats_port = args[14]
-    self.heron_internals_config_file = args[15]
+        base64.b64decode(parsed_args.instance_jvm_opts.lstrip('"').
+                         rstrip('"').replace('&equals;', '='))
+    self.classpath = parsed_args.classpath
+    self.master_port = parsed_args.master_port
+    self.tmaster_controller_port = parsed_args.tmaster_controller_port
+    self.tmaster_stats_port = parsed_args.tmaster_stats_port
+    self.heron_internals_config_file = parsed_args.heron_internals_config_file
     self.component_rammap =\
-        map(lambda x: {x.split(':')[0]: int(x.split(':')[1])}, args[16].split(','))
+        map(lambda x: {x.split(':')[0]:
+                           int(x.split(':')[1])}, parsed_args.component_rammap.split(','))
```

Comment:
I was wondering if this logic could be encapsulated into the argparse logic via custom actions. I'm not certain it could, and maybe it's not worth the effort, so feel free to disregard.